### PR TITLE
[GEOS-11081] Add option to disable GetFeatureInfo transforming raster layers

### PR DESCRIPTION
**Pull request by sikeoka**
[GEOS-11081](https://osgeo-org.atlassian.net/browse/GEOS-11081)

This PR adds a WMS service setting and a vendor option to SLD FeatureTypeStyle elements to disable applying rendering transformations to raster data for WMS GetFeatureInfo requests.

This PR replaces https://github.com/geoserver/geoserver/pull/7021 which contains some additional discussion.

I would like to get this into the 2.22.5 release if possible since I am supporting customers still using Java 8.

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by sikeoka** [comment]
@aaime, do you have any additional comments about this? I put your comments into the documentation.

File: src/web/wms/src/main/java/org/geoserver/wms/web/WMSAdminPage.html [hunk 1]
```diff
@@ -244,6 +244,20 @@
       </ul>
     </fieldset>
 
+    <fieldset>
+      <legend><span><wicket:message key="featuresTransformDisabledTitle"></wicket:message></span></legend>
+      <ul class="choiceList">
+        <li>
+          <input type="checkbox" wicket:id="featuresTransformDisabled"></input>
```
Comment (aaime) [inline src/web/wms/src/main/java/org/geoserver/wms/web/WMSAdminPage.html:251]: "features" is out of context and confusing:
* The flag disables _rendering_ transformations
* It (currently) works only on rasters (but could be implemented for features too, later, although that's going to be harder)

Could it be called "renderingTransformDisabledTitle" instead, and do the same changes in other places that are currently using "featuresTransform"? (e.g., WMSInfo and friends).

**Review by aaime — CHANGES_REQUESTED** [review]

File: src/web/wms/src/main/java/org/geoserver/wms/web/WMSAdminPage.html (hunk 1 above)
Comment (sikeoka) [reply src/web/wms/src/main/java/org/geoserver/wms/web/WMSAdminPage.html]: I changed transformFeatures to transformFeatureInfo.

**Review by aaime — APPROVED** [review]

**Comment by aaime** [comment]
Makes sense, looks good, merging!

**Comment by aaime** [comment]
The backport to `2.22.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 2e18c2ddeb... [GEOS-11081] Add option to disable GetFeatureInfo transforming raster layers
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/web/wms/src/main/java/org/geoserver/wms/web/WMSAdminPage.html
CONFLICT (content): Merge conflict in src/web/wms/src/main/java/org/geoserver/wms/web/WMSAdminPage.html
Auto-merging src/wms/src/main/java/org/geoserver/wms/WMS.java

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-2.22.x 2.22.x
# Navigate to the new working tree
cd .worktrees/backport-2.22.x
# Create a new branch
git switch --create backport-7041-to-2.22.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 2e18c2ddebd2c2cb98fae29c5f52bbf16eb6c245
# Push it to GitHub
git push --set-upstream origin backport-7041-to-2.22.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-2.22.x
```
Then, create a pull request where the `base` branch is `2.22.x` and the `compare`/`head` branch is `backport-7041-to-2.22.x`.

**Pull request merged**